mod-audit-server/src/main/java/org/folio/util/marc/ParsedRecordUtil.java

```diff
@@ -0,0 +1,243 @@
+      result.merge(tag, formattedValue, (existing, newValue) -> {
+        if (existing instanceof List) {
+          ((List<Object>) existing).add(newValue);
+          return existing;
+        }
+        return new ArrayList<>(Arrays.asList(existing, newValue));
+      });
+    }));
+    return result;
+  }
+
+  /**
+   * Formats the given field value into a standardized string representation.
+   * If the input value is a map containing subfields, it creates a formatted string
+   * combining indicators and subfields. Otherwise, it converts the value to a string.
+   *
+   * @param value the field value to format, which can be an object or a map with subfields
+   *              and indicators ("ind1", "ind2").
+   * @return a formatted string representation of the field.
+   */
+  @SuppressWarnings("unchecked")
+  private static String formatField(Object value) {
+    if (value instanceof Map<?, ?> map && map.containsKey(SUBFIELDS_KEY)) {
+      var fieldData = (Map<String, Object>) map;
+      var ind1 = (String) fieldData.getOrDefault("ind1", " ");
+      var ind2 = (String) fieldData.getOrDefault("ind2", " ");
+      var subfields = (List<Map<String, Object>>) fieldData.get(SUBFIELDS_KEY);
+      var subfieldsString = subfields.stream()
+        .flatMap(subObj -> subObj.entrySet().stream()
+          .map(e -> "$" + e.getKey() + e.getValue()))
+        .collect(Collectors.joining());
+      return ind1 + ind2 + subfieldsString;
+    }
+    return value.toString();
+  }
+
+  /**
+   * Compares two maps representing parsed records and identifies added, modified,
+   * and removed entries. It determines the differences by comparing the keys
+   * and values of the given maps.
+   *
+   * @param oldMap the original map representing the state of a parsed record before changes
+   * @param newMap the updated map representing the state of a parsed record after changes
+   * @return a map containing three keys ("added", "removed", "modified"), each mapped
+   * to a list of maps. The "added" key contains entries present only in the new map,
+   * the "removed" key contains entries present only in the old map,
+   * and the "modified" key contains entries with mismatched values between the maps.
+   */
+  private static Map<String, Object> compareParsedRecords(Map<String, Object> oldMap, Map<String, Object> newMap) {
+    List<Map<String, Object>> added = new ArrayList<>();
+    List<Map<String, Object>> modified = new ArrayList<>();
+    List<Map<String, Object>> removed = new ArrayList<>();
+
+    newMap.forEach((key, newValue) -> {
+      if (!oldMap.containsKey(key)) {
+        added.add(Map.of(FIELD_KEY, key, VALUE_KEY, newValue));
+      }
+    });
+
+    oldMap.forEach((key, oldValue) -> {
+      if (!newMap.containsKey(key)) {
+        removed.add(Map.of(FIELD_KEY, key, VALUE_KEY, oldValue));
+      }
+    });
+
+    newMap.forEach((key, newValue) -> {
+      if (!"005".equals(key) && oldMap.containsKey(key)) {
```

Comment:
@dmytrokrutii  Please extract "005" field and "oldValue"/"newValue" to the constants. This is business-logic-related values.